### PR TITLE
Add minimumParticipants,value,period to FrameworkAgreement

### LINKED ISSUE #129: Techniques: Clarify its use case in the readme
i.e. `tender.techniques.frameworkAgreement.method` describes the available methods, known at the time of setting up the framework agreement.

(Regarding https://github.com/open-contracting/standard/issues/784, to indicate which method was actually used for a specific call-off, we need another field.)

We can also add synonyms for the methods, i.e. when the FA is being set up, it is communicated in advance whether the FA will use direct call-offs (withoutReopeningCompetition), mini-competitions (withReopeningCompetition) or both (withAndWithoutReopeningCompetition).

### PR DESCRIPTION
**Pull request by ColinMaudry**
Closes https://github.com/open-contracting-extensions/european-union/issues/79
Closes  https://github.com/open-contracting-extensions/european-union/issues/80
Closes https://github.com/open-contracting-extensions/european-union/issues/84
Closes https://github.com/open-contracting/ocds-extensions/issues/129

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,6 +80,10 @@ Report issues for this extension in the [ocds-extensions repository](https://git
 
 ## Changelog
 
+### 2020-10-05
+
+* Add `minimumParticipants` field to `frameworkAgreement`.
```
Comment (jpmckinney) [inline README.md]:
```suggestion
* Add `FrameworkAgreement.minimumParticipants` field.
```

File: release-schema.json [hunk 2]
```diff
@@ -111,6 +119,16 @@
...
           "minLength": 1
+        },
+        "value": {
+          "title": "Value",
+          "description": "The total upper estimated value of the framework agreement.",
+          "$ref": "#/definitions/Value"
+        },
+        "period": {
+          "title": "Period",
+          "description": "The period when the framework agreement is active.",
```
Comment (jpmckinney) [inline release-schema.json]: Moving in line with `tender/contractPeriod`.

```suggestion
          "description": "The period over which the framework agreement is estimated to be active.",
```

File: README.md [hunk 3]
```diff
@@ -79,6 +87,10 @@ Report issues for this extension in the [ocds-extensions repository](https://git
 
 ## Changelog
 
+### 2020-10-05
+
+* Add `minimumParticipants`, `value` fields to `frameworkAgreement`
```
Comment (jpmckinney) [inline README.md]:
```suggestion
* Add `minimumParticipants`, `value` and `period` fields to the `FrameworkAgreement` object.
```

**Comment by jpmckinney** [comment]
Noting that "The total upper estimated value of the framework agreement" is based on the description of `tender/value`.

**Comment by ColinMaudry** [comment]
I'll add https://github.com/open-contracting-extensions/european-union/issues/84 to this PR.

**Comment by ColinMaudry** [comment]
I've tried to find a good example for `frameworkAgreement.description`, in line with the mapping guidance we've given to the UK. Tell me if we'd better use a `<placeholder>`.

```
"description": "Call offs are estimated to be organized every 3 months, with an average value of 60,000 euros per contract."
```

**Comment by ColinMaudry** [comment]
I've tried to find a good example for `frameworkAgreement.description`, in line with the mapping guidance we've given to the UK. Tell me if we'd better use a <placeholder>.

```
"description": "Call offs are estimated to be organized every 3 months, with an average value of 60,000 euros per contract."
```

File: release-schema.json [hunk 4]
```diff
@@ -129,6 +129,14 @@
           "title": "Period",
           "description": "The period when the framework agreement is active.",
           "$ref": "#/definitions/Period"
+        },
+        "description": {
+          "title": "Description",
+          "description": "A summary description of the framework agreement. This complements any structured information provided using the other fields. Descriptions should be short and easy to read. Avoid using ALL CAPS.",
```
Comment (jpmckinney) [inline release-schema.json]: Let's limit the description to semantics and remove guidance on composition.

```suggestion
          "description": "A summary description of the framework agreement. This complements any structured information provided using the other fields.",
```

**Review by jpmckinney — COMMENTED** [review]
The `description` example looks good!

**Comment by jpmckinney** [comment]
@ColinMaudry Should we add guidance explaining that the `tender.value` and `tender.contractPeriod` fields are preferred, and that the `value` and `period` fields on the `tender.frameworkAgreement` object should only be used if a data source provides values and periods for each of the contract and framework agreement, like TED XML Schema R2.08?

**Comment by ColinMaudry** [comment]
@jpmckinney It would consequently make sense to use `Lot.techniques.frameworkAgreement.value` (or `period`) if there are more that one lot, but not to use a single `Tender.techniques.frameworkAgreement.value`, as `tender.value` would be more appropriate if the process has only one `FrameworkAgreement`?

**Comment by ColinMaudry** [comment]
I'll also try to tackle https://github.com/open-contracting/ocds-extensions/issues/129 and add the synonyms (withReopeningCompetition = mini-competition, etc.).

**Comment by ColinMaudry** [comment]
Tell me what you think: https://github.com/open-contracting-extensions/ocds_techniques_extension/pull/7/files#diff-04c6e90faac2675aa89e2176d2eec7d8R9

Closes https://github.com/open-contracting/ocds-extensions/issues/129

**Comment by jpmckinney** [comment]
> @jpmckinney It would consequently make sense to use `Lot.techniques.frameworkAgreement.value` (or `period`) if there are more that one lot, but not to use a single `Tender.techniques.frameworkAgreement.value`, as `tender.value` would be more appropriate if the process has only one `FrameworkAgreement`?

Aha, right. I think we want:

> The `value` and `period` fields of `FrameworkAgreement` objects should only be used if a data source provides values and periods for both the contract/lot and the framework agreement, like TED XML Schema R2.08. Otherwise:
> 
> * If a procurement isn't divided into lots, use the `tender.value` and `tender.contractPeriod` fields.
> * If a procurement is divided into lots, use the `value` and `contractPeriod` fields of `Lot` objects.

**Comment by jpmckinney** [comment]
@ColinMaudry Is this ready for review again? Looks good to me.

**Comment by ColinMaudry** [comment]
@jpmckinney It's ready now

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**